src/utils/components/SSHAccess/components/SSHCommand.tsx

```diff
@@ -38,22 +35,37 @@ const SSHCommand: React.FC<SSHCommandProps> = ({
   sshServiceLoaded,
 }) => {
   const { t } = useKubevirtTranslation();
-  const [sshService, setSSHService] = useState<IoK8sApiCoreV1Service>();
-  const { command, user, sshServiceRunning } = useSSHCommand(vm, sshService);
+  const [sshService, setSSHService] = useState<IoK8sApiCoreV1Service | null>();
+  const { command, sshServiceRunning } = useSSHCommand(vm, sshService);
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState<Error>();
 
-  const onSSHChange = (enableSSH: boolean) => {
+  const onSSHChange = async (newServiceType: SERVICE_TYPES) => {
     setLoading(true);
 
-    const request = enableSSH
-      ? createSSHService(vm, vmi).then(setSSHService)
-      : deleteSSHService(sshService).then(() => setSSHService(undefined));
+    try {
+      const deletePromise = sshService
+        ? deleteSSHService(sshService)
+        : new Promise<void>((resolve) => resolve());
 
-    request.catch(setError).finally(() => setLoading(false));
-  };
+      const createPromise =
+        !newServiceType || newServiceType === SERVICE_TYPES.NONE
+          ? new Promise((resolve) => resolve(null))
+          : createSSHService(vm, vmi, newServiceType);
 
-  const hasNoUsername = user === CLOUD_INIT_MISSING_USERNAME;
+      await Promise.allSettled([deletePromise, createPromise]).then(
+        ([deleteResult, createResult]) => {
```

Comment:
why do u combine them? create and delete and the same logic? its 2 different actions.

Reply:
Yes, i would not wait for delete to start the creation request. But okay, I'll wait first for the delete and then create